### DISCUSSION TITLE
connected_regions_allowed

### DISCUSSION THREAD

**Discussion by uganzedo1974** [CliMA/ClimaOcean.jl#279] [Q&A]
Hello;
Good job with this package, it's excellent. I have a question, in some example in the regrid_bathymetry option there was the connected_regions_allowed option that is now not available. I would like to see the possibility of doing a downscaling in a simulation of a certain area. In an area, increase the simulation resolution. Is this possible? Thank you for your work
Unai

**Comment by simone-silvestri** [comment]
Hello @uganzedo1974. The `connected_regions_allowed` option has been changed to `major_basins`. 
For example:
```julia
bottom = regrid_bathymetry(grid; connected_regions_allowed = 2)
```
with the new syntax is
```julia
bottom = regrid_bathymetry(grid; major_basins = 2)
```
That keyword argument, however, might have been a misnomer before. It was not there to specify the number of connected domains in the simulation but to specify the number of "independent major basins", or fluid regions fully encompassed by land, that are retained in the bathymetric map. This is probably much better expressed with the `major_basins` keyword argument. There is an example in the examples folder (`examples/generate_bathymetry.jl`) that shows the effect of this keyword argument.

Concerning downscaling certain areas, we cannot run coupled nested simulations yet because the open boundary conditions required to connect different domains have not yet been implemented in Oceananigans.

**Comment by uganzedo1974** [comment]
Thank you for your quick response. Maybe you can give me an idea to solve an issue with your package.
I have seen that there is a good example of downscaling in the Mediterranean area, which is a closed sea and the contour limits are the coast.
But I would like to do a study of the Canary Islands and I have done so but the contour limits are ocean, so u and v rotate in that contour limit, something that is not realistic. Is there any way to eliminate that condition?
Thanks again
Unai

**Comment by simone-silvestri** [comment]
I think what you need to have are open boundary conditions that allow you to prescribe boundary conditions for the velocity field from a "parent" large-scale simulation. This is the typical procedure when doing regional simulations. 
However, to do so you would need an implementation of open boundary conditions in Oceananigans (the parent package of ClimaOcean) which at the moment is missing. Therefore, realistically, you can only do simulations that do not involve inflows or outflows. 

A way you could try solving this issue would be to make your domain as large as possible (such that boundary conditions do not influence excessively your region of interest) and prescribe a strong restoring to climatology near the lateral boundaries. You do this using the ECCORestoring functionality. 
https://github.com/CliMA/ClimaOcean.jl/blob/19c5eff9b15313f8c786c107c5d77661b1dab244/src/DataWrangling/ECCO/ECCO_restoring.jl#L242-L307

This is demonstrated in the Mediterranean example in #244. I can copy here the code:
https://github.com/CliMA/ClimaOcean.jl/blob/e1ed35f62638244feb2ee7b0a57913a093b6e00d/examples/mediterranean_simulation_with_ecco_restoring.jl#L69-L88

You can also take a look at the docstring by doing in julia by doing:
```julia
julia> ? ClimaOcean.ECCO.ECCORestoring
```

You probably also need to implement a sponge region for velocities, you can take a look at this example to see how:
https://github.com/CliMA/ClimaOcean.jl/blob/19c5eff9b15313f8c786c107c5d77661b1dab244/experiments/mesoscale_resolving_omip/lat_lon_near_global_simulation.jl#L47-L60

**Comment by glwagner** [comment]
Should we move this to a discussion?
